scrapy/core/engine.py

```diff
@@ -269,7 +269,7 @@ def _downloaded(
     ) -> Union[Deferred, Response]:
         assert self.slot is not None  # typing
         self.slot.remove_request(request)
-        return self.download(result, spider) if isinstance(result, Request) else result
+        return self.download(result) if isinstance(result, Request) else result
```

Comment:
If I recall right, @elacuesta was working on this, and the reason that argument remains there is for backward compatibility (_downloaded is private, but callers of _downloaded that are not private pass the `spider` argumetn and that argument should be passed).

I do understand that self.download must be logging a warning when called with that parameter. Maybe we could make it so that calls that come from _downloaded do not cause the warning.